XenAdmin/Core/Updates.cs

```diff
@@ -649,6 +649,66 @@ public static List<XenServerPatch> GetMinimalPatches(IXenConnection conn, XenSer
             return minimalPatches;
         }
 
+        /// <summary>
+        /// Gets all the patches for the given connection
+        /// </summary>
+        public static List<XenServerPatch> GetAllPatches(IXenConnection conn)
+        {
+            var version = GetCommonServerVersionOfHostsInAConnection(conn, XenServerVersions);
+            return GetAllPatches(version);
+        }
+
+        /// <summary>
+        /// Gets an upgrade sequence that contains a version upgrade, optionally followed by all the patches for the new version
+        /// </summary>
+        public static List<XenServerPatch> GetAllPatches(IXenConnection conn, XenServerPatchAlert alert, bool updateTheNewVersion)
+        {
+            Debug.Assert(conn != null);
+            Debug.Assert(alert != null);
+
+            var allPatches = new List<XenServerPatch> { alert.Patch };
+
+            // if it's a version updgrade the update sequence will be this patch (the upgrade) and the patches for the new version
+            if (updateTheNewVersion && alert.NewServerVersion != null && alert.NewServerVersion.MinimalPatches != null)

```

Comment:
Shouldn't the last bit be `alert.NewServerVersion.Patches != null`? If the MinimalPatches are null, we'll miss the patches from line 691 too. Or perhaps removed that bit completely and only AddRange if GetAllPatches(alert.NewServerVersion) returns non-null?